runtime/startup_nat.c

```diff
@@ -155,7 +157,9 @@ value caml_startup_common(char_os **argv, int pooling)
     if (caml_termination_hook != NULL) caml_termination_hook(NULL);
     return Val_unit;
```

Comment:
Shouldn't you also call `caml_terminate_signals` in this code path ?

Reply:
Well spotted!  This code path corresponds to the initial thread calling `Thread.exit`.  The `caml_termination_hook` will cause the process to terminate.  But even in this case it's worth freeing the alternate stack in case a memory leak detector is being used (like they do at Redhat).  

Come to think of it, there's also `Stdlib.exit` !  I added a call to `caml_terminate_signals` in this case too.